airflow/contrib/kubernetes/kubernetes_request_factory/kubernetes_request_factory.py

```diff
@@ -131,15 +131,44 @@ def extract_volume_secrets(pod, req):
 
     @staticmethod
     def extract_env_and_secrets(pod, req):
-        env_secrets = [s for s in pod.secrets if s.deploy_type == 'env']
-        if len(pod.envs) > 0 or len(env_secrets) > 0:
+        envs_from_key_secrets = [
+            env for env in pod.secrets if env.deploy_type == 'env' and hasattr(env, 'key')
+        ]
+
+        envs_from_secrets = [
+            env for env in pod.secrets if env.deploy_type == 'env' and not hasattr(env, 'key')
+        ]
+
+        if pod.envs_from_configmaps or envs_from_secrets:
+            req['spec']['containers'][0]['envFrom'] = []
+
+        if len(pod.envs) > 0 or len(envs_from_key_secrets) > 0:
             env = []
             for k in pod.envs.keys():
                 env.append({'name': k, 'value': pod.envs[k]})
-            for secret in env_secrets:
+            for secret in envs_from_key_secrets:
                 KubernetesRequestFactory.add_secret_to_env(env, secret)
+
             req['spec']['containers'][0]['env'] = env
 
+        for secret in envs_from_secrets:
```

Comment:
This seems like a LOT happening in a single function. Would it be possible to create some helper functions to make this more readable?

Reply:
@dimberman In first version of this PR I had this in seperate functions, but @ashb suggested to move it here. Could we agree what is the best so I don't change this back and forth again? :)

Reply:
@ashb any thoughts on this? Perhaps we could move it into helper functions but keep it in the same class?

Reply:
@dimberman No preference :)

Reply:
@galuszkak int that case please seperate some of this logic into private helper functions. After that should be good to merge :)

Reply:
To be clear - this  means making `extract_env_and_secrets` call extra helper functions, so no new changes outside of  kube_request_factory.py

Reply:
@dimberman @ashb changes applied, it's fixed! Hope this is OK now.

Thanks, for a lot of feedback!